neqo-http3/src/connection_client.rs

```diff
@@ -472,7 +472,27 @@ impl Http3Client {
     }
 
     fn handle_goaway(&mut self, goaway_stream_id: u64) -> Res<()> {
-        qinfo!([self], "handle_goaway");
+        qinfo!([self], "handle_goaway {}", goaway_stream_id);
+
+        let id = StreamId::from(goaway_stream_id);
+        if id.is_uni() || id.is_server_initiated() {
+            return Err(Error::HttpId);
+        }
+
+        match self.base_handler.state {
+            Http3State::Connected => {
+                self.base_handler.state = Http3State::GoingAway(goaway_stream_id);
+            }
+            Http3State::GoingAway(ref mut stream_id) => {
+                if goaway_stream_id > *stream_id {
+                    return Err(Error::HttpId);
```

Comment:
Might be worth a new code for this.

Reply:
Also, you don't test this branch.  You should.

Reply:
yes, I need test, sorry it slipped.
Te spec does not define an error for this. I want to write a spec issue (It was just too late for me yesterday)